pyccl/core.py

```diff
@@ -634,9 +649,19 @@ def compute_distances(self):
         """Compute the distance splines."""
         if self.has_distances:
             return
-        status = 0
-        status = lib.cosmology_compute_distances(self.cosmo, status)
-        check(status, self)
+        if not self.background_on_input:
+            status = 0
+            status = lib.cosmology_compute_distances(self.cosmo, status)
+            check(status, self)
+        else:
+            # TODO: Check that chi_z and z_array are arrays of the same size.
+            # TODO: Check that z_array has the correct order.
+            status = 0
+            a = 1./(1+self.z_array)
```

Comment:
Yes indeed, I created z decreasing. I suppose it would be good to check if the input a_array is indeed monotonically increasing at some point in the code, perhaps here.